feat(web, web-react): add popoverProps, labelProps to Picker #DS-2568 (#2672)
lmc-eu/spirit-design-system: packages/web-react/src/components/UNSTABLE_Picker/README.md

- **`popoverProps.theme`** — dropdown panel with options.

On a **Light on Brand** surface, keep the label on-brand and use **Light Default** for the field and popover:

```tsx
<Box theme="theme-light-on-brand" backgroundColor="primary" padding="space-800" borderRadius="300">
  <UNSTABLE_UncontrolledPicker
    id="demo-picker-themes"
    defaultSelectedKeys={['cs', 'dk']}
    label="Languages"
    theme="theme-light-default"
    labelProps={{ theme: 'theme-light-on-brand' }}
    popoverProps={{ theme: 'theme-light-default' }}
  >
    {/* … */}
  </UNSTABLE_UncontrolledPicker>
</Box>
```

See the [Themes demo][picker-themes-demo].

### `dropdownProps`, `popoverProps`, and `labelProps`

Forward extra props to the inner `Dropdown`, `DropdownPopover`, and `Label`.

| Prop            | Target            | Example                      |
| --------------- | ----------------- | ---------------------------- |
| `dropdownProps` | `Dropdown`        | `fullWidthMode`, `placement` |
| `popoverProps`  | `DropdownPopover` | `theme`, `data-testid`       |
| `labelProps`    | `Label`           | `theme`                      |

> Reserved props are set by the picker after spreading. For `dropdownProps` and `labelProps`, these props are omitted from the types. For `popoverProps`, do not pass reserved attributes such as `data-spirit-fullwidthmode` or `data-spirit-placement`, even though generic `data-*` attributes are still allowed by the current typing — use `fullWidthMode` and `placement` on `dropdownProps` instead.

Reviewer: ```suggestion
Reserved props are set by the picker after spreading.

⚠️ For `dropdownProps` and `labelProps`, reserved props are omitted from the types.

⚠️ For `popoverProps`, do not pass reserved attributes such as `data-spirit-fullwidthmode` or `data-spirit-placement`, even though generic `data-*` attributes are still allowed by the current typing — use `fullWidthMode` and `placement` on `dropdownProps` instead.
```

| `id`                  | `string`                                                  | —                                | ✓        | Stable id for the picker and related elements                                                                             |
| `isOpen`              | `bool`                                                    | —                                | ✓        | Popover open state                                                                                                        |
| `label`               | `string`                                                  | —                                | ✓        | Visible label and accessible name for the control                                                                         |
| `dropdownProps`       | `UnstablePickerDropdownProps`                             | —                                | ✕        | Props for the inner `Dropdown` (for example `fullWidthMode`, `placement`)                                                 |

> | `labelProps`          | `UnstablePickerLabelProps`                                | —                                | ✕        | Props for the inner `Label` (for example `theme`)                                                                         |

Reviewer: Suggested change
      
    
    
      
          
            
            | `dropdownProps`       | `UnstablePickerDropdownProps`                             | —                                | ✕        | Props for the inner `Dropdown` (for example `fullWidthMode`, `placement`)                                                 |
          
          
            
            | `labelProps`          | `UnstablePickerLabelProps`                                | —                                | ✕        | Props for the inner `Label` (for example `theme`)                                                                         |
          
          
            
            | `dropdownProps`       | `UnstablePickerDropdownProps`                             | —                                | ✕        | Props for the inner `Dropdown`, see  [`dropdownProps`, `popoverProps`, and `labelProps`](#dropdownprops-popoverprops-and-labelprops)                                                 |
          
          
            
            | `labelProps`          | `UnstablePickerLabelProps`                                | —                                | ✕        | Props for the inner `Label`, see  [`dropdownProps`, `popoverProps`, and `labelProps`](#dropdownprops-popoverprops-and-labelprops)                                                                         |

| `onToggle`            | `() => void`                                              | —                                | ✓        | Toggle callback; parent updates `isOpen`                                                                                  |
| `onSelectionChange`   | `(keys: string[]) => void`                                | —                                | ✓        | Called when the selection changes                                                                                         |

> | `popoverProps`        | `UnstablePickerPopoverProps`                              | —                                | ✕        | Safe props for the inner `DropdownPopover` (for example `theme`); reserved props are omitted from the type                |

Reviewer: ```suggestion
| `popoverProps`        | `UnstablePickerPopoverProps`                              | —                                | ✕        | Safe props for the inner `DropdownPopover`, see  [`dropdownProps`, `popoverProps`, and `labelProps`](#dropdownprops-popoverprops-and-labelprops)                |
```

| `size`                | [Size dictionary][dictionary-size]                        | `medium`                         | ✕        | Size of the picker shell                                                                                                  |
| `tagDescriptionText`  | `string`                                                  | i18n `picker.tagDescriptionText` | ✕        | Hidden text for screen readers (tag removal hint)                                                                         |
| `hasValidationIcon`   | `bool`                                                    | `false`                          | ✕        | Whether to show the validation icon                                                                                       |
| `validationState`     | [Validation dictionary][dictionary-validation]            | —                                | ✕        | Validation state                                                                                                          |

> | `validationText`      | `ReactNode` \| `ReactNode[]`                              | —                                | ✕        | Validation message                                                                                                        |

Reviewer: 🔤
